src/machine/machine_atsam.go

```diff
@@ -0,0 +1,31 @@
+//go:build sam
+
+package machine
+
+import (
+	"runtime/volatile"
+	"unsafe"
+)
+
+// HardwareID returns an identifier that is unique within
+// a particular chipset.
+//
+// The identity is one burnt into the MCU itself, or the
+// flash chip at time of manufacture.
+//
+// It's possible that two different vendors may allocate
+// the same HardwareID, so callers should take this into
+// account if needing to generate a globally unique id.
+//
+// The length of the hardware ID is vendor-specific, but
+// 8 bytes (64 bits) and 16 bytes (128 bits) are common.
+func HardwareID() []byte {
+	id := make([]byte, 16)
```

Comment:
Perhaps avoid the heap allocation here?

```
var hardwareID [16]byte

func HardwareID() []byte {
    ...
    return hardwareID[:]
}
```

Reply:
thanks for reviewing.  yeah, i like that change - will update the PR